### PR TITLE
WIP: build: arm32/arm64 build support.

### PR DESCRIPTION
**Pull request by PokeMMO**
```
#Add support for multiarch environments.
dpkg --add-architecture armhf
dpkg --add-architecture arm64

#install toolchains
apt install gcc-arm-linux-gnueabihf g++-arm-linux-gnueabihf gcc-aarch64-linux-gnu g++-aarch64-linux-gnu

#Afterward you can install dependencies for other archs like
apt install libgl1-mesa-dev:arm64 libgl1-mesa-dev:armhf
```

TODO:
- [ ] cleanup the conditions
- [ ] document required dependencies
- [ ] fix all builds and flags
- [ ] get travis to build arm32/arm64
- [ ] add `CROSSBUILD` or similar flag to enable/disable gXX prefixes.
- [ ] replace suffixes with folder per arch?
- [ ] SharedLibraryLoader support.

### CODE REVIEW

File: config/linux/build.xml [hunk 1]
```diff
@@ -5,25 +5,63 @@
...
+       <not>
+          <isset property="LIB_POSTFIX"/>
+       </not>
+    </condition>
+
     <condition property="gcc.suffix" value="-${gcc.version}" else="">
         <isset property="gcc.version"/>
     </condition>
 
+    <condition property="gcc.preffix" value="arm-linux-gnueabihf-">
```
Comment (intrigus) [inline config/linux/build.xml]: Typo. It's 'prefix' not 'preffix'

File: config/linux/build.xml [hunk 2]
```diff
@@ -5,25 +5,63 @@
...
+
+    <condition property="LIB_POSTFIX" value="arm32">
+        <istrue value="${build.arch.arm32}"/>
+    </condition>
+
+    <condition property="LIB_POSTFIX" value="arm64">
+        <istrue value="${build.arch.arm64}"/>
+    </condition>
+
+    <condition property="LIB_POSTFIX" value="">
```
Comment (intrigus) [inline config/linux/build.xml]: In the rest of this file the word suffix is used instead, e.g. Line 63&64.
Might be good to have some consistency.

Comment (PokeMMO) [reply config/linux/build.xml]: Tentative plan (as discussed in slack) was to remove postfix entirely and do per-folder archs. (Similar to android)

File: config/linux/build.xml (hunk 1 above)
Comment (PokeMMO) [reply config/linux/build.xml]: Thanks

**Comment by PokeMMO** [comment]
Current version will now build all modules except the following: (Edit config/build-bindings.xml to test)

```
    <property name="binding.llvm" value="false"/> <!-- arm32 fails, arm64 fine -->
    <property name="binding.meow" value="false"/> <!-- Missing arm32 values https://github.com/LWJGL/lwjgl3/blob/master/modules/lwjgl/meow/src/main/c/meow_intrinsics.h, arm64 still fails for other reasons -->
    <property name="binding.nfd" value="false"/> <!-- libgtk-3-dev can't be installed for multiple archs... -->
    <property name="binding.openvr" value="false"/> <!-- No arm -->
    <property name="binding.sse" value="false"/> <!-- SSE -->
    <property name="binding.tootle" value="false"/> <!-- JMLSSEVec, hard dep on sse -->

**Comment by Spasi** [comment]
Thank you @PokeMMO!

Summary of final changes:

- Kept a single `build.xml` for all Linux builds. I wanted to avoid duplication and the current version is reasonably clean.
- Cross compilation is enabled with the already supported `LWJGL_BUILD_ARCH` environment variable. E.g. `export LWJGL_BUILD_ARCH=arm64`.
- Shared libraries now live in per-architecture subfolders and are named identically (i.e. dropped the `32` suffix from Windows x86 binaries). This applies to both the Ant build and the natives artifacts for deployment.
- Introduced the `Platform.getArchitecture` API.
- Fixed the LLVM bindings for arm32 (was a generator bug) and meow's build flags for arm64.
- Refactored LWJGL-CI to use a matrix of jobs for Linux & macOS. This makes it much simpler to maintain and add new platforms/architectures in the future. The script that builds `lwjgl3` is [here](https://github.com/LWJGL-CI/lwjgl3/blob/master/.travis.yml).

**Pull request merged**